code/src/components/ShowProduct.js

```diff
@@ -0,0 +1,122 @@
+import React from 'react'
+import { useDispatch, useSelector } from 'react-redux';
+import { barcodeReducer } from 'reducers/barcodeReducer'
+import { PieChart } from 'components/PieChart';
+import styled from 'styled-components';
+import { Button } from 'lib/Buttons';
+
+export const ShowProduct = () => {
+  const product = useSelector(store => store.reducer.product)
+  const { product_name: name } = product
+  const { fat_100g: fat, carbohydrates_100g: carb, sugars_100g: sugar, proteins_100g: protein } = product.nutriments
+  const dispatch = useDispatch();
+
+  const saveCurrent = () => {
+    dispatch(barcodeReducer.actions.saveProduct(product))
+  }
+
+  return (
+    <Wrapper>
+      <div>
+        <H3>{name}</H3>
+        <table>
+          <tbody>
+            <Tr>
+              <Th>Nutrients</Th>
+              <Th>Per 100g</Th>
+            </Tr>
+            <Tr>
+              <Td>Calories</Td>
+              <Td>{product.nutriments['energy-kcal_value'] ? `${product.nutriments['energy-kcal_value']} kcal` : '-'}</Td>
+            </Tr>
+            <Tr>
+              <Td>Fat</Td>
+              <Td>{fat ? `${fat} g` : '-'}</Td>
+            </Tr>
+            <Tr>
+              <Td>Carbohydrates</Td>
+              <Td>{carb ? `${carb} g` : '-'}</Td>
+            </Tr>
+            <Tr>
+              <Td>Sugars</Td>
+              <Td>{sugar ? `${sugar} g` : '-'}</Td>
+            </Tr>
+            <Tr>
+              <Td>Fiber</Td>
+              <Td>{product.nutriments.fiber_100g ? `${product.nutriments.fiber_100g} g` : '-'}</Td>
+            </Tr>
+            <Tr>
+              <Td>Proteins</Td>
+              <Td>{protein ? `${protein} g` : '-'}</Td>
+            </Tr>
+          </tbody>
+        </table>
```

Comment:
Is it possible to extract this table in to its own component, as you did with the pie chart?

Reply:
Absolutely, that would be a lot cleaner :)